api/src/services/lottery.service.ts

```diff
@@ -224,4 +257,325 @@ export class LotteryService {
 
     return ordinalArray;
   }
+
+  /**
+   *
+   * @param queryParams
+   * @param req
+   * @returns generates the lottery export file via helper function and returns the streamable file
+   */
+  async lotteryExport<QueryParams extends ApplicationCsvQueryParams>(
+    req: ExpressRequest,
+    res: Response,
+    queryParams: QueryParams,
+  ): Promise<StreamableFile> {
+    const user = mapTo(User, req['user']);
+    await this.authorizeLotteryExport(user, queryParams.listingId);
+
+    const workbook = new Excel.Workbook();
+
+    const filename = join(
+      process.cwd(),
+      `src/temp/lottery-listing-${queryParams.listingId}-applications-${
+        user.id
+      }-${new Date().getTime()}.xlsx`,
+    );
+
+    const zipFilePath = join(
+      process.cwd(),
+      `src/temp/lottery-listing-${queryParams.listingId}-applications-${
+        user.id
+      }-${new Date().getTime()}.zip`,
+    );
+
+    await this.createLotterySheet(workbook, {
+      ...queryParams,
+      includeDemographics: true,
+    });
+
+    await workbook.xlsx.writeFile(filename);
+
+    const readStream = createReadStream(filename);
+
+    return new Promise((resolve) => {
+      // Create a writable stream to the zip file
+      const output = fs.createWriteStream(zipFilePath);
+      const archive = archiver('zip', {
+        zlib: { level: 9 },
+      });
+      output.on('close', () => {
+        const zipFile = createReadStream(zipFilePath);
+        resolve(new StreamableFile(zipFile));
+      });
+
+      archive.pipe(output);
+      archive.append(readStream, {
+        name: `lottery-${queryParams.listingId}-${new Date().getTime()}.xlsx`,
+      });
+      archive.finalize();
+    });
+  }
+
+  /**
+   *
+   * @param filename
+   * @param queryParams
+   * @returns generates the lottery sheet
+   */
+  async createLotterySheet<QueryParams extends ApplicationCsvQueryParams>(
+    workbook: Excel.Workbook,
+    queryParams: QueryParams,
+  ): Promise<void> {
+    let applications = await this.prisma.applications.findMany({
+      select: {
+        id: true,
+        preferences: true,
+        householdMember: {
```

Comment:
could you just get householdSize rather than going to the householdMember table? If I'm reading this correctly you just need this to determine the headers

Reply:
thats a good question, this is how the old csv export was calculating it so I kept that logic in tact 

But after looking at prod data with:
```

SELECT
    SUM(1)
FROM applications app1
    JOIN (
        SELECT
            a.id,
            SUM(1) as sm
        FROM applications a
            LEFT JOIN household_member hm ON hm.application_id = a.id
        GROUP BY a.id
    ) app2 ON app2.id = app1.id
        AND COALESCE(app2.sm, 0) != COALESCE(app1.household_size, 0);


SELECT
    SUM(1)
FROM applications app1
    JOIN (
        SELECT
            a.id,
            SUM(1) as sm
        FROM applications a
            LEFT JOIN household_member hm ON hm.application_id = a.id
        GROUP BY a.id
    ) app2 ON app2.id = app1.id
        AND COALESCE(app2.sm, 0) = COALESCE(app1.household_size, 0);
        ```
        
        what I'm seeing is that household_size doesn't seem to always match up with the actual house hold size so switching that would probably cause problems

Reply:
Interesting. Okie dokie lets keep it as the householdmembers instead of the size then. I wonder if the inconsistency is due to a bug or just older data isn't accurate.